pkg/tls/cred.go

```diff
@@ -90,6 +91,14 @@ func (crt *Crt) Verify(roots *x509.CertPool, name string) error {
 	}
 	vo := x509.VerifyOptions{Roots: roots, Intermediates: i, DNSName: name}
 	_, err := crt.Certificate.Verify(vo)
+
+	//If error due to expiry
+	if err == fmt.Errorf("x509: certificate has expired or is not yet valid") {
```

Comment:
Is there a better way to check if the error is due to expiry?

Reply:
Very good question!. So the verify method returns a special type of error which is [`CertificateInvalidError`](https://github.com/golang/go/blob/master/src/crypto/x509/verify.go#L72) It looks like: 

```
type CertificateInvalidError struct {
	Cert   *Certificate
	Reason InvalidReason
	Detail string
}
```

What you can do is to check whether the error returned is of this type and if it is check the `Reason` field. If it equals to `Expired` and there is a type for that already, you can assume that this is the reason. Also this time validity of the cert can be bad not only due to expiration. It can be the case that this cert is still not valid (interesting why they named the const Expired...) Anyhow, that is all to say that maybe in your messag you should include the NotBefore as well.